src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,417 @@
+                MemberExpression property = Expression.Property(param, key);
+                object value = item.Value;
+
+                Expression compare = null;
+                ODataEnumValue enumValue = value as ODataEnumValue;
+                if (enumValue != null)
+                {
+                    value = enumValue.Value;
+                }
+                Expression constant = parameterizeConstant ? LinqParameterContainer.Parameterize(value.GetType(), value) : Expression.Constant(value);
+                if (directionMap.ContainsKey(key))
+                {
+                    compare = directionMap[key] == OrderByDirection.Descending ? binder.CreateBinaryExpression(BinaryOperatorKind.LessThan, property, constant, true) : binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+                else
+                {
+                    compare = binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+
+                if (firstProperty)
+                {
+                    lastEquality = binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true);
+                    where = compare;
+                    firstProperty = false;
+                }
+                else
+                {
+                    Expression condition = Expression.AndAlso(lastEquality, compare);
+                    where = where == null ? condition : Expression.OrElse(where, condition);
+                    lastEquality = Expression.AndAlso(lastEquality, binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true));
+                }
+            }
+
+            Expression whereLambda = Expression.Lambda(where, param);
+            return ExpressionHelpers.Where(query, whereLambda, query.ElementType);
+        }
+
+        /// <summary>
+        /// Generates a dictionary with property name and property values specified in the skiptoken value.
+        /// </summary>
+        /// <param name="value">The skiptoken string value.</param>
+        /// <param name="context">The <see cref="ODataQueryContext"/> which contains the <see cref="IEdmModel"/> and some type information</param>
+        /// <returns>Dictionary with property name and property value in the skiptoken value.</returns>
+        private static IDictionary<string, object> PopulatePropertyValuePairs(string value, ODataQueryContext context)
+        {
+            Contract.Assert(context != null);
+
+            IDictionary<string, object> propertyValuePairs = new Dictionary<string, object>();
+            IList<string> keyValuesPairs = ParseValue(value, CommaDelimiter);
+            foreach (string pair in keyValuesPairs)
+            {
+                string[] pieces = pair.Split(new char[] { propertyDelimiter }, 2);
+                if (pieces.Length > 1 && !String.IsNullOrWhiteSpace(pieces[0]))
```

Comment:
>if (pieces.Length > 1 && !String.IsNullOrWhiteSpace(pieces[0])) [](start = 16, length = 63)

What is this logic trying to achieve? Is this handling whitespace? Couldn't that be done in ParseValue?

Reply:
Defensively coding for bad nextlinks as discussed. Will keep it open for now. 

---
In reply to: [293042120](https://github.com/OData/WebApi/pull/1724#discussion_r293042120) [](ancestors = 293042120)

Reply:
Should throw and exception here - Mike!

---
In reply to: [293089961](https://github.com/OData/WebApi/pull/1724#discussion_r293089961) [](ancestors = 293089961,293042120)